assets/src/data/model/status/constants.js

```diff
@@ -7,46 +12,67 @@ export const STATUS_TYPE_PAYMENT = 'payment';
-export const STATUS_ID_EMAIL_DRAFT = 'EDR';
-export const STATUS_ID_EMAIL_SENT = 'ESN';
-export const STATUS_ID_EMAIL_EXPIRED = 'EXP';
+export const emailStatus = {
+	DRAFT: 'EDR',
+	SENT: 'ESN',
+	EXPIRED: 'EXP',
+};
 // event
-export const STATUS_ID_EVENT_ACTIVE = 'ACT';
-export const STATUS_ID_EVENT_REGISTRATION_CLOSED = 'CLS';
-export const STATUS_ID_EVENT_DELETED = 'DEL';
-export const STATUS_ID_EVENT_DENIED = 'DEN';
-export const STATUS_ID_EVENT_DRAFT = 'DRF';
-export const STATUS_ID_EVENT_NOT_ACTIVE = 'NAC';
-export const STATUS_ID_EVENT_REGISTRATION_NOT_OPEN = 'NOP';
-export const STATUS_ID_EVENT_ONGOING = 'ONG';
-export const STATUS_ID_EVENT_REGISTRATION_OPEN = 'OPN';
-export const STATUS_ID_EVENT_PENDING = 'PND';
-export const STATUS_ID_EVENT_SECONDARY = 'SEC';
+export const eventStatus = {
+	ACTIVE: 'ACT',
+	REGISTRATION_CLOSED: 'CLS',
+	DELETED: 'DEL',
+	DENIED: 'DEN',
+	DRAFT: 'DRF',
+	NOT_ACTIVE: 'NAC',
+	NOT_OPEN: 'NOP',
+	ONGOING: 'ONG',
+	REGISTRATION_OPEN: 'OPN',
+	PENDING: 'PND',
+	SECONDARY: 'SEC',
+};
 // message
-export const STATUS_ID_MESSAGE_DEBUG = 'MDO';
-export const STATUS_ID_MESSAGE_EXECUTING = 'MEX';
-export const STATUS_ID_MESSAGE_FAIL = 'MFL';
-export const STATUS_ID_MESSAGE_INCOMPLETE = 'MIC';
-export const STATUS_ID_MESSAGE_IDLE = 'MID';
-export const STATUS_ID_MESSAGE_RESEND = 'MRS';
-export const STATUS_ID_MESSAGE_SENT = 'MSN';
+export const messageStatus = {
+	DEBUG: 'MDO',
+	EXECUTING: 'MEX',
+	FAIL: 'MFL',
+	INCOMPLETE: 'MIC',
+	IDLE: 'MID',
+	RESEND: 'MRS',
+	SENT: 'MSN',
+};
 // payment
-export const STATUS_ID_PAYMENT_APPROVED = 'PAP';
-export const STATUS_ID_PAYMENT_CANCELLED = 'PCN';
-export const STATUS_ID_PAYMENT_DECLINED = 'PDC';
-export const STATUS_ID_PAYMENT_FAILED = 'PFL';
-export const STATUS_ID_PAYMENT_PENDING = 'PPN';
+export const paymentStatus = {
+	APPROVED: 'PAP',
+	CANCELLED: 'PCN',
+	DECLINED: 'PDC',
+	FAILED: 'PFL',
+	PENDING: 'PPN',
+};
 // registration
-export const STATUS_ID_REGISTRATION_APPROVED = 'RAP';
-export const STATUS_ID_REGISTRATION_CANCELLED = 'RCN';
-export const STATUS_ID_REGISTRATION_DECLINED = 'RDC';
-export const STATUS_ID_REGISTRATION_INCOMPLETE = 'RIC';
-export const STATUS_ID_REGISTRATION_NOT_APPROVED = 'RNA';
-export const STATUS_ID_REGISTRATION_PENDING_PAYMENT = 'RPP';
-export const STATUS_ID_REGISTRATION_WAIT_LIST = 'RWL';
+export const registrationStatus = {
+	APPROVED: 'RAP',
+	CANCELLED: 'RCN',
+	DECLINED: 'RDC',
+	INCOMPLETE: 'RIC',
+	NOT_APPROVED: 'RNA',
+	PENDING_PAYMENT: 'RPP',
+	WAIT_LIST: 'RWL',
+};
 // transaction
-export const STATUS_ID_TRANSACTION_ABANDONED = 'TAB';
-export const STATUS_ID_TRANSACTION_COMPLETE = 'TCM';
-export const STATUS_ID_TRANSACTION_FAILED = 'TFL';
-export const STATUS_ID_TRANSACTION_INCOMPLETE = 'TIN';
-export const STATUS_ID_TRANSACTION_OVERPAID = 'TOP';
+export const transactionStatus = {
+	ABANDONED: 'TAB',
+	COMPLETE: 'TCM',
+	FAILED: 'TFL',
+	INCOMPLETE: 'TIN',
+	OVERPAID: 'TOP',
+};
+
+export const ALL_STATUSES = [
```

Comment:
for consistency... do we want to use ALL CAPS for a constant that is not a single value?

Or should the constants above for the status ID arrays be capitalized ??? 

ie: `EVENT_STATUS_ARRAY` 

as well... since there are filters now for JS, maybe we should slap a filter on this?

Reply:
Convention in javascript is that usage of `ALL_CAPS` is a signal that the value is intended to be immutable because `const` in javascript simply prevents reassigning (not immutability).  So it's acceptable to use CAPS in this case as a signal that it is considered immutable.  I didn't capitalize the individual model status arrays because I figured they are an object representation that _may_ be mutable (i.e. new temporary status added to the object).  However the individual object keys ARE all caps to signal immutability for the strings.

Based on that description though, I'm open to changing `emailStatus` to `EMAIL_STATUS`.  This would have the added benefit of still allowing client code to do something like:

```
let emailStatus = EMAIL_STATUS.SENT
```

> FWIW, I think the *_STATUS_ID_ARRAY schema is the most accurate and most consistent format, but I'll let you decide in case you think of something even better.

As mentioned earlier I disagree for all the reasons I already gave in my previous comment.  They are objects intended for easily accessing the individual statuses (similar to constant).

> One final thought... since there are filters now for JS, maybe we should slap a filter on this array so it can be modified at runtime?

Ya good thought, however within javascript I'd prefer to really be hardline on not implementing a filter and/or action until we've got an actual use-case implementing it.  Reason being is we _may_ find that the implementation of the filter/action might end up being done differently than we initially thought (so as to limit its scope if necessary etc).

Reply:
kk good points... 

do we still want to be pedantic about this being an object containing status **ID**s, ie: `EMAIL_STATUS_ID` vs `EMAIL_STATUS`  ???

I'll leave that and whether or not to imply immutability by capitalizing the object names up to you.